src/main/groovy/com/bmuschko/gradle/docker/tasks/image/DockerBuildImage.groovy

```diff
@@ -18,8 +18,10 @@ package com.bmuschko.gradle.docker.tasks.image
 import com.bmuschko.gradle.docker.DockerRegistryCredentials
 import com.bmuschko.gradle.docker.tasks.AbstractDockerRemoteApiTask
 import com.bmuschko.gradle.docker.tasks.RegistryCredentialsAware
+import com.bmuschko.gradle.docker.utils.TaskStateHelper
 import org.gradle.api.tasks.*
 
+
 class DockerBuildImage extends AbstractDockerRemoteApiTask implements RegistryCredentialsAware {
```

Comment:
Instead of shoving this logic into this class I'd create a new class and delegate from there. That'll separate logic, make it more readable and maintainable.